js/lightbeam.js

```diff
@@ -1,9 +1,33 @@
 async function renderGraph() {
-  const canvas = document.getElementById('canvas');
-  const context = canvas.getContext('2d');
   const websites = await store.getAll();
 
-  viz.draw(context, websites);
+  // reformat data for D3
```

Comment:
**jonathan Kingston:**
This looks like it is coupled to the visualisations. It shouldn't be.
This code might:
    - Change hostname paths and clean replicated data
    - Query first party and third parties in a an order it chooses
It shouldn't be changing output to match how a rendering might display it.

**biancadanforth**:
What are you proposing I do here instead? We don't want data from the store to worry about which visualization library is used, but the data needs to be of a certain format for drawing in D3 (i.e. it expects an array of objects, not an object of nested objects).
I agree with you that the order in which it assembles `nodes_data` and `links_data` could be different from the order those keys appear in `websites`, but why does that matter?

**jonathanKingston**:
By ordering I meant async nature. I could imagine this code querying the data in such a way that it draws in order of last used or most common or some other factor.
I'm now thinking the structure passed to vis might be better expressed as individual websites.
If you are manipulating data to match the nodes being drawn it likely best belongs where it is about to be drawn.
Now that I have thought about the code a little more I thin there are a few avenues:
1. Vis code could accept big blobs and reshuffle itself, however that binds the vis code to the nested structure of how he storage currently works (Long term I think this is a little cumbersome so likely we shouldn't do this)
2. This code could call individual methods on viz like viz.drawSite(hostname, {icon: "..", other: things}) and viz.connectSite(hostname, thirdPartyHostname)
3. Invent a new structure to pass to viz, this suffers from a similar issue to 1.
4. seems the most clean, however I am happy to merge this which I think currently is using 3. I think 2. would allow a little more flexibility for viz to draw much faster in the future and this code is in charge of what is drawn first.